drivers/i2c/ccs811_driver.go

```diff
@@ -0,0 +1,349 @@
+	l := &CCS811Driver{
+		name:      gobot.DefaultName("CCS811"),
+		connector: a,
+		measMode:  NewCCS811MeasMode(),
+		//Recommended resistance value is 100,000
+		ntcResistanceValue: 100000,
+		Config:             NewConfig(),
+	}
+
+	for _, option := range options {
+		option(l)
+	}
+
+	return l
+}
+
+//WithCCS811MeasMode sets the sampling rate of the device
+func WithCCS811MeasMode(mode CCS811DriveMode) func(Config) {
+	return func(c Config) {
+		d, _ := c.(*CCS811Driver)
+		d.measMode.driveMode = mode
+	}
+}
+
+//WithCCS811NTCResistance sets reistor value used in the temperature calculations.
+//This resistor must be placed between pin 4 and pin 8 of the chip
+func WithCCS811NTCResistance(val uint32) func(Config) {
+	return func(c Config) {
+		d, _ := c.(*CCS811Driver)
+		d.ntcResistanceValue = val
+	}
+}
+
+//Start initializes the sensor
+func (d *CCS811Driver) Start() (err error) {
+	bus := d.GetBusOrDefault(d.connector.GetDefaultBus())
+	address := d.GetAddressOrDefault(ccs811DefaultAddress)
+
+	if d.connection, err = d.connector.GetConnection(address, bus); err != nil {
+		return err
+	}
+
+	return d.initialize()
+}
+
+//Name returns the Name for the Driver
+func (d *CCS811Driver) Name() string { return d.name }
+
+//SetName sets the Name for the Driver
+func (d *CCS811Driver) SetName(n string) { d.name = n }
+
+//Connection returns the connection for the Driver
+func (d *CCS811Driver) Connection() gobot.Connection { return d.connector.(gobot.Connection) }
+
+//Halt returns true if devices is halted successfully
+func (d *CCS811Driver) Halt() (err error) { return }
+
+//GetHardwareVersion returns the hardware version of the device in the form of 0x1X
+func (d *CCS811Driver) GetHardwareVersion() (uint8, error) {
+	v, err := d.connection.ReadByteData(ccs811RegHwVersion)
+	if err != nil {
+		return 0, err
+	}
+
+	return v, nil
+}
+
+//GetFirmwareBootVersion returns the bootloader version
+func (d *CCS811Driver) GetFirmwareBootVersion() (uint16, error) {
+	v, err := d.connection.ReadWordData(ccs811RegFwBootVersion)
+	if err != nil {
+		return 0, err
+	}
+
+	return v, nil
+}
+
+//GetFirmwareAppVersion returns the app code version
+func (d *CCS811Driver) GetFirmwareAppVersion() (uint16, error) {
+	v, err := d.connection.ReadWordData(ccs811RegFwAppVersion)
+	if err != nil {
+		return 0, err
+	}
+
+	return v, nil
+}
+
+//GetStatus returns the current status of the device
+func (d *CCS811Driver) GetStatus() (*CCS811Status, error) {
+	s, err := d.connection.ReadByteData(ccs811RegStatus)
+	if err != nil {
+		return nil, err
+	}
+
+	cs := NewCCS811Status(s)
+	return cs, nil
+}
+
+//GetTemperature returns the device temperature in celcius.
+//If you do not have an NTC resistor installed, this function should not be called
+func (d *CCS811Driver) GetTemperature() (float32, error) {
+
+	buf, err := d.read(ccs811RegNtc, 4)
+	if err != nil {
+		return 0, err
+	}
+
+	vref := ((uint16(buf[0]) << 8) | uint16(buf[1]))
```

Comment:
I grabbed the needed calculation from [here](https://github.com/adafruit/Adafruit_CircuitPython_CCS811/blob/master/adafruit_ccs811.py#L165)